### PR TITLE
docs: update docs to show role selectors

### PR DESCRIPTION
**Pull request by debs-obrien**

### CODE REVIEW

**Review by yury-s — APPROVED** [review]

File: docs/src/writing-tests-csharp.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ public class UnitTest1 : PageTest
         await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));
 
         // create a locator
-        var getStarted = Page.Locator("text=Get Started");
+        var getStarted = Page.GetByRole(AriaRole.Link, new () { Name = "Get started" });
```
Comment (yury-s) [inline docs/src/writing-tests-csharp.md:74]: style nit: I saw Max fixing options code to be formatted as `new() {` instead of `new () {`

**Review by yury-s — APPROVED** [review]

File: docs/src/writing-tests-csharp.md (hunk 1 above)
Comment (mxschmitt) [inline docs/src/writing-tests-csharp.md:74]:
```suggestion
        var getStarted = Page.GetByRole(AriaRole.Link, new() { NameString = "Get started" });
```

File: docs/src/writing-tests-csharp.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));
 [Locators](./locators.md) are the central piece of Playwright's auto-waiting and retry-ability. Locators represent a way to find element(s) on the page at any moment and are used to perform actions on elements such as `.ClickAsync` `.FillAsync` etc. Custom locators can be created with the [`method: Page.locator`] method.
 
 ```csharp
-var getStarted = Page.Locator("text=Get Started");
+var getStarted = Page.GetByRole(AriaRole.Link, new () { Name = "Get started" });
```
Comment (mxschmitt) [inline docs/src/writing-tests-csharp.md:105]:
```suggestion
var getStarted = Page.GetByRole(AriaRole.Link, new() { NameString = "Get started" });
```

File: docs/src/writing-tests-csharp.md [hunk 3]
```diff
@@ -111,7 +111,7 @@ await getStarted.ClickAsync();
 [Selectors](./selectors.md) are strings that are used to create Locators. Playwright supports many different selectors like [Text](./selectors.md#text-selector), [CSS](./selectors.md#css-selector), [XPath](./selectors.md#xpath-selectors) and many more. Learn more about available selectors and how to pick one in this [in-depth guide](./selectors.md).
 
 ```csharp
-await Expect(Page.Locator("text=Installation")).ToBeVisibleAsync();
+await Expect(Page.GetByRole(AriaRole.Heading, new () { Name = "Installation" })).ToBeVisibleAsync();
```
Comment (mxschmitt) [inline docs/src/writing-tests-csharp.md:114]:
```suggestion
await Expect(Page.GetByRole(AriaRole.Heading, new() { NameString = "Installation" })).ToBeVisibleAsync();
```

**Comment by debs-obrien** [comment]
Can we make sure codegen also writes this as I took example from codgen

File: docs/src/writing-tests-csharp.md [hunk 4]
```diff
@@ -36,7 +36,7 @@ public class Tests : PageTest
         await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));
 
         // create a locator
-        var getStarted = Page.Locator("text=Get Started");
+        var getStarted = Page.GetByRole(AriaRole.Link, new () { Name = "Get started" });
```
Comment (mxschmitt) [inline docs/src/writing-tests-csharp.md:39]:
```suggestion
        var getStarted = Page.GetByRole(AriaRole.Link, new() { NameString = "Get started" });
```

**Review by mxschmitt — APPROVED** [review]
one nit, otherwise lgtm

**Pull request merged**